src/librustdoc/doctest.rs

```diff
@@ -468,6 +472,13 @@ fn run_test(
                 return Err(TestFailure::UnexpectedRunPass);
             } else if !should_panic && !out.status.success() {
                 return Err(TestFailure::ExecutionFailure(out));
+            } else if options.nocapture {
+                if !out.stdout.is_empty() {
+                    io::stdout().write_all(&out.stdout).expect("failed to write stdout");
+                }
+                if !out.stderr.is_empty() {
+                    io::stderr().write_all(&out.stderr).expect("failed to write stderr");
+                }
```

Comment:
Just a question: why check if they are empty before writing out?

Reply:
To prevent useless calls. Not sure if very useful though...

Reply:
According to the docs, `write_all` immediately returns if passed an empty input, so 🤷

Reply:
`write_all` does, but we still need to get lock from `io::stdout()` and `io::stderr()` (that's the main reason behind these conditions :wink: ).

Reply:
Based on the docs and the source code, it seems that `io::stdout()` doesn't actually lock stdout; instead, it is locked upon writing.

Reply:
Absolutely but it creates a few things to do so: https://doc.rust-lang.org/nightly/src/std/io/stdio.rs.html#552-559

Anyway, if you prefer me to remove the checks, I'll do so. Where does your preference go? :)

Reply:
I think I lean on the side of removing the checks just to make the code a bit simpler :)

Reply:
Na it's fine, let's go for simpler code!